File: backend/src/main/java/ru/chsergeig/shoppy/service/common/CommonOrdersService.java
package ru.chsergeig.shoppy.service.common;

import ru.chsergeig.shoppy.dto.ExtendedOrderDto;
import ru.chsergeig.shoppy.dto.admin.GoodDto;
import ru.chsergeig.shoppy.dto.admin.OrderDto;

import java.util.List;

public interface CommonOrdersService {

    String createOrder(List<GoodDto> goods, String username);

    ExtendedOrderDto getOrderByGuid(String guid, String username);

}

File: backend/src/main/kotlin/ru/chsergeig/shoppy/service/common/CommonOrdersService.kt
package ru.chsergeig.shoppy.service.common

import ru.chsergeig.shoppy.model.CommonGoodDto
import ru.chsergeig.shoppy.model.ExtendedOrderDto

interface CommonOrdersService {

    fun createOrder(
        goods: List<CommonGoodDto?>?,
        username: String?
    ): String

    fun getOrderByGuid(
        guid: String?,
        username: String?
    ): ExtendedOrderDto
}
